In [1]:
import sys
import os

project_root = os.path.abspath('..')
if project_root not in sys.path:
    sys.path.append(project_root)

%load_ext autoreload
%autoreload 2


In [2]:
import pandas as pd
import numpy as np
import polars as pl

import matplotlib.pyplot as plt
import seaborn as sns

from src.data import make_dataset
from src.features import build_features
from src.visualization import visualize

sns.set_style()
plt.rcParams["figure.figsize"] = (12, 6)

HOSPITAL_A_ANLIZAR = 112103  # Este es el Torax

## 1. Obtencion de Métricas de Egresos Hospitalarios

En este apartado se quieren calculcar diversas métricas relevantes para el análisis de los
egresos hospitalarios en Chile. Las métricas a calcular son las siguientes:

1. Cantidad de Egresos Hospitalarios
2. Dias de Estada Totales
3. Cantidad de Intervenciones Quirúrgicas
4. Cantidad de Muertos

Todas estas métricas serán calculadas para cada Año, Establecimiento de Salud y Diagnóstico
distinto que exista en la base de datos. Por ejemplo:

"En el año 2019, el Hospital San José (código XXXXXX) tuvo XX egresos, con XX dias estada promedio,
con XX intervenciones quirúrgicas y XX muertos para el diagnóstico A24.4"s


In [3]:
df_nacional = pl.scan_csv(
    "../data/processed/egresos_procesados.csv",
    infer_schema_length=10000,
    null_values="Extranjero",
)

In [4]:
AGRUPACION = [
    "ANO_EGRESO",
    "ESTABLECIMIENTO_SALUD",
    "GLOSA_ESTABLECIMIENTO_SALUD",
    "DIAG1",
]

metricas = build_features.obtener_metricas_egresos(df_nacional, AGRUPACION).collect()

In [5]:
display(metricas.sample(10))

shape: (10, 8)
┌────────────┬──────────────┬──────────────┬───────┬───────────┬─────────────┬─────────┬───────────┐
│ ANO_EGRESO ┆ ESTABLECIMIE ┆ GLOSA_ESTABL ┆ DIAG1 ┆ n_egresos ┆ dias_estada ┆ n_int_q ┆ n_muertos │
│ ---        ┆ NTO_SALUD    ┆ ECIMIENTO_SA ┆ ---   ┆ ---       ┆ _totales    ┆ ---     ┆ ---       │
│ i64        ┆ ---          ┆ LUD          ┆ str   ┆ u32       ┆ ---         ┆ i64     ┆ i64       │
│            ┆ i64          ┆ ---          ┆       ┆           ┆ i64         ┆         ┆           │
│            ┆              ┆ str          ┆       ┆           ┆             ┆         ┆           │
╞════════════╪══════════════╪══════════════╪═══════╪═══════════╪═════════════╪═════════╪═══════════╡
│ 2003       ┆ 103219       ┆ Hospital del ┆ K719  ┆ 1         ┆ 4           ┆ 0       ┆ 0         │
│            ┆              ┆ Cobre        ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Salvador     ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Alle…        ┆       ┆           ┆             ┆         ┆           │
│ 2015       ┆ 125100       ┆ Hospital     ┆ J358  ┆ 1         ┆ 2           ┆ 1       ┆ 0         │
│            ┆              ┆ Regional     ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ (Coihaique)  ┆       ┆           ┆             ┆         ┆           │
│ 2008       ┆ 111200       ┆ Hospital     ┆ T888  ┆ 1         ┆ 2           ┆ 0       ┆ 0         │
│            ┆              ┆ Clínico      ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Universidad  ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Cat…         ┆       ┆           ┆             ┆         ┆           │
│ 2014       ┆ 101100       ┆ Hospital Dr. ┆ M663  ┆ 3         ┆ 6           ┆ 3       ┆ 0         │
│            ┆              ┆ Juan Noé     ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Crevanni (…  ┆       ┆           ┆             ┆         ┆           │
│ …          ┆ …            ┆ …            ┆ …     ┆ …         ┆ …           ┆ …       ┆ …         │
│ 2004       ┆ 117201       ┆ Clínica      ┆ D175  ┆ 1         ┆ 3           ┆ 1       ┆ 0         │
│            ┆              ┆ Chillán Más  ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ Vida         ┆       ┆           ┆             ┆         ┆           │
│ 2004       ┆ 109100       ┆ Complejo     ┆ N484  ┆ 1         ┆ 3           ┆ 0       ┆ 0         │
│            ┆              ┆ Hospitalario ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ San José (…  ┆       ┆           ┆             ┆         ┆           │
│ 2002       ┆ 116100       ┆ Hospital San ┆ O926  ┆ 3         ┆ 3           ┆ 3       ┆ 0         │
│            ┆              ┆ Juan de Dios ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ (Curic…      ┆       ┆           ┆             ┆         ┆           │
│ 2003       ┆ 107109       ┆ Hospital     ┆ I691  ┆ 1         ┆ 40          ┆ 0       ┆ 0         │
│            ┆              ┆ Juana Ross   ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ de Edwards   ┆       ┆           ┆             ┆         ┆           │
│            ┆              ┆ (…           ┆       ┆           ┆             ┆         ┆           │
└────────────┴──────────────┴──────────────┴───────┴───────────┴─────────────┴─────────┴───────────┘

- Al ver una muestra del calculo realizado, se puede observar la cantidad de egresos, dias de
  estada, numero de intervenciones quirurgicas y cantidad de muertos para 10 duplas de
  hospitales - diagnosticos distintos.


Si bien se tienen las metricas calculadas, ahora se quiere responder a las siguientes preguntas:

- En qué diagnósticos se destaca cada hospital productivamente? Esto significa, en qué diagnósticos
  cada hospital tiene una mayor cantidad de egresos que otros hospitales?

- El hospital tiene un desempeño similar dentro de distintos grupos/estratos de hospitales?
  El hospital se desempeñara de la misma forma nacionalmente, dentro del estrato público, privado,
  etc...?

Para responder a la pregunta se realizará un ranking de todos los hospitales para cada
diagnóstico y cada año observado. Este ranking se realizará para distintos grupos/estratos de
hospitales (Nacional, Hospitales Públicos, Hospitales Privados, Hospitales GRD)


## 2. Análisis de Estratos

En este análisis se quiere observar el ranking de cada hospital/clínica del país para cada 
año y diagnóstico observado. Además, el ranking se realizará en los siguientes estratos:

1. Nacional
2. Públicos
3. Privados
4. Adscritos a GRD
5. INT

In [6]:
dict_estratos = build_features.obtener_diccionario_estratos(df_nacional, HOSPITAL_A_ANLIZAR)

In [35]:
variables_a_rankear = ["n_egresos"]
subgrupo_del_ranking = ["ANO_EGRESO", "DIAG1"]

rankings_nacionales = build_features.agregar_ranking_estratos(
    metricas, dict_estratos, variables_a_rankear, subgrupo_del_ranking
)

rankings_hospital_analizado = rankings_nacionales.filter(
    pl.col("ESTABLECIMIENTO_SALUD") == HOSPITAL_A_ANLIZAR
)

In [61]:
display(rankings_nacionales.head(8))

shape: (8, 27)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ ANO_EGRES ┆ ESTABLECI ┆ GLOSA_EST ┆ Capítulo  ┆ … ┆ %_grd_n_e ┆ ranking_i ┆ total_int ┆ %_intern │
│ O         ┆ MIENTO_SA ┆ ABLECIMIE ┆ ---       ┆   ┆ gresos    ┆ nterno_n_ ┆ erno_n_eg ┆ o_n_egre │
│ ---       ┆ LUD       ┆ NTO_SALUD ┆ str       ┆   ┆ ---       ┆ egresos   ┆ resos     ┆ sos      │
│ i64       ┆ ---       ┆ ---       ┆           ┆   ┆ f64       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ i64       ┆ str       ┆           ┆   ┆           ┆ u32       ┆ u32       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ 2020      ┆ 104103    ┆ Hospital  ┆ Cap.21    ┆ … ┆ 1.0       ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Provincia ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ l del     ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Huasco M… ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 125102    ┆ Hospital  ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Dr.       ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Leopoldo  ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Ortega R. ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ …         ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 125103    ┆ Hospital  ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Lord      ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Cochrane  ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 118106    ┆ Hospital  ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de Lota   ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 116101    ┆ Hospital  ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de Teno   ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 107107    ┆ Hospital  ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de        ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Petorca   ┆ QUE       ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ INFLUYEN  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ EN…       ┆   ┆           ┆           ┆           ┆          │
│ 2020      ┆ 102200    ┆ Clínica   ┆ Cap.21    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Iquique   ┆ FACTORES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ QUE     

Por lo tanto, se pueden observar todas las métricas de egresos hospitalarios nacionales para
cada estrato definido. Además, se observa que en casos de empate en cantidad de egresos,
el ranking es repetido.

## 3. Diagnósticos más relevantes del Instituto Nacional del Tórax

En este apartado se quieren responder las siguientes preguntas:

1. ¿Cuáles son los **diagnósticos más relevantes productivamente** en el Instituto Nacional del
Tórax?
2. ¿En qué diagnósticos el INT tiene una mayor cantidad de Egresos que cualquier otro establecimiento
de salud Nacional? En otras palabras, ¿En qué diagnósticos el **INT tiene un Ranking 1 Nacional**?

Estas preguntas se quieren responder tanto para diagnósticos ambulatorios (n_int_q = 0) y 
quirúrgicos (n_int_q > 0).

Además, se sabe que desde el 2011 el registro de Egresos Hospitalarios DEIS fue llevado por la
unidad de Grupo Relacionado al Diagnóstico (GRD). Debido a lo anterior, la calidad de la codificación
de diagnósticos es mucho mayor. Tomado de lo anterior, **se obtendrán los diagnósticos que tengan
una alta productividad desde el 2012 hasta 2020**.

In [73]:
DIAGS_MAS_RELEVANTES_QUIRURGICOS = rankings_hospital_analizado.filter(
    (pl.col("ranking_nacionales_n_egresos") == 1)
    & (pl.col("n_int_q") > 0)
    & (pl.col("ANO_EGRESO") > 2011)
).sort(by=["ANO_EGRESO", "n_egresos", "n_int_q"], descending=True)

veinte_mas_relevantes_por_anio = DIAGS_MAS_RELEVANTES_QUIRURGICOS.group_by(["ANO_EGRESO"]).head(20)

In [74]:
DIAGS_MAS_RELEVANTES_QUIRURGICOS = veinte_mas_relevantes_por_anio.select(
    pl.col("DIAG1").unique()
).sort(by="DIAG1")

In [78]:
print(list(DIAGS_MAS_RELEVANTES_QUIRURGICOS.to_series()))
print(f"Son {len(DIAGS_MAS_RELEVANTES_QUIRURGICOS)} diagnosticos quirurgicos")

['C33X', 'C340', 'C341', 'C342', 'C343', 'C381', 'C384', 'C450', 'C780', 'C782', 'D143', 'D381', 'E848', 'I080', 'I081', 'I340', 'I350', 'I351', 'I352', 'I420', 'I456', 'I472', 'I495', 'I710', 'I712', 'J398', 'J47X', 'J679', 'J841', 'J848', 'J849', 'J860', 'J869', 'J90X', 'J91X', 'J931', 'J955', 'J980', 'M348', 'Q211', 'Q231', 'Q676', 'T820', 'T821', 'Z450']
Son 45 diagnosticos quirurgicos


In [40]:
DIAGS_1_o_2_en_2019 = list((
    rankings_hospital_analizado.filter(
        (pl.col("DIAG1").is_in(DIAGS_MAS_RELEVANTES.to_series()))
        & (pl.col("ANO_EGRESO") == 2019)
        & (
            (pl.col("ranking_nacionales_n_egresos") == 1)
            | (pl.col("ranking_nacionales_n_egresos") == 2)
        )
    )
    .select(pl.col("DIAG1"))
).to_series())

DIAGS_1_o_2_en_2019 += ["T821", "I710", "I051", "D143"]

In [41]:
print(len(DIAGS_1_o_2_en_2019))

40


In [42]:
tabla_resumen_egresos_torax_relevantes = (
    rankings_hospital_analizado.filter(
        ((pl.col("ANO_EGRESO") >= 2016) & (pl.col("ANO_EGRESO") <= 2019))
        & (pl.col("DIAG1").is_in(DIAGS_1_o_2_en_2019))
    )
    .with_columns((pl.col("DIAG1") + " - " + pl.col("Descripción")).alias("Diagnóstico"))
    .to_pandas()
    .pivot(
        index=["Diagnóstico"],
        columns="ANO_EGRESO",
        values=[
            "ranking_nacionales_n_egresos",
            "n_egresos",
            "total_nacionales_n_egresos",
            "%_nacionales_n_egresos",
        ],
    )
).sort_values(
    by=[("ranking_nacionales_n_egresos", 2019), ("n_egresos", 2019)], ascending=[True, False]
)

In [43]:
resumen_egresos_torax_y_nacional_relevantes = (
    tabla_resumen_egresos_torax_relevantes[("n_egresos")].astype(int).astype(str)
    + " ("
    + tabla_resumen_egresos_torax_relevantes[("total_nacionales_n_egresos")].astype(int).astype(str)
    + "; "
    + round((tabla_resumen_egresos_torax_relevantes[("%_nacionales_n_egresos")] * 100), 1).astype(
        str
    )
    + "%)"
)

posiciones_torax_egresos_relevantes = tabla_resumen_egresos_torax_relevantes[
    ("ranking_nacionales_n_egresos")
].astype(int)

resumen_egresos_y_posiciones_relevantes = pd.concat(
    [resumen_egresos_torax_y_nacional_relevantes, posiciones_torax_egresos_relevantes], axis=1
)

resumen_egresos_y_posiciones_relevantes.columns = [
    "2016",
    "2017",
    "2018",
    "2019",
    "2016",
    "2017",
    "2018",
    "2019",
]

display(resumen_egresos_y_posiciones_relevantes)

2016  \
Diagnóstico                                                            
C341 - Tumor maligno del lóbulo superior, bronq...  180 (400; 45.0%)   
Z450 - Asistencia y ajuste de marcapaso cardíaco    101 (1146; 8.8%)   
J841 - Otras enfermedades pulmonares interstici...  124 (1617; 7.7%)   
C343 - Tumor maligno del lóbulo inferior, bronq...  120 (241; 49.8%)   
D381 - Tumor de comportamiento incierto o desco...   91 (637; 14.3%)   
J679 - Neumonitis debida a hipersensibilidad a ...    53 (68; 77.9%)   
C780 - Tumor maligno secundario del pulmón           43 (328; 13.1%)   
J848 - Otras enfermedades pulmonares interstici...    40 (98; 40.8%)   
I351 - Insuficiencia (de la válvula) aórtica          19 (209; 9.1%)   
J955 - Estenosis subglótica consecutiva a proce...   36 (124; 29.0%)   
I080 - Trastornos de las válvulas mitral y aórtica   35 (112; 31.2%)   
I081 - Trastornos de las válvulas mitral y tric...    43 (64; 67.2%)   
J47X - Bronquiectasia                                58 (370; 15.7%)   
E848 - Fibrosis quística con otras manifestaciones    23 (51; 45.1%)   
C782 - Tumor maligno secundario de la pleura         42 (136; 30.9%)   
I340 - Insuficiencia (de la válvula) mitral           39 (392; 9.9%)   
I352 - Estenosis (de la válvula) aórtica con in...    35 (94; 37.2%)   
I456 - Síndrome de preexcitación                     31 (190; 16.3%)   
M348 - Otras formas de esclerosis sistémica           17 (37; 45.9%)   
I472 - Taquicardia ventricular                        14 (356; 3.9%)   
T820 - Complicación mecánica de prótesis de vál...     8 (39; 20.5%)   
J869 - Piotórax sin fístula                           19 (334; 5.7%)   
C342 - Tumor maligno del lóbulo medio, bronquio...    25 (53; 47.2%)   
C381 - Tumor maligno del mediastino anterior           9 (26; 34.6%)   
I712 - Aneurisma de la aorta torácica, sin menc...    13 (94; 13.8%)   
J860 - Piotórax con fístula                           15 (49; 30.6%)   
C384 - Tumor maligno de la pleura                     17 (55; 30.9%)   
C33X - Tumor maligno de la tráquea                     4 (20; 20.0%)   
I495 - Síndrome del seno enfermo                      68 (756; 9.0%)   
T821 - Complicación mecánica de dispositivo ele...   20 (157; 12.7%)   
I420 - Cardiomiopatía dilatada                        26 (463; 5.6%)   
D143 - Tumor benigno de los bronquios y del pulmón   21 (189; 11.1%)   
J849 - Enfermedad pulmonar intersticial, no esp...    36 (556; 6.5%)   
I710 - Disección de aorta (cualquier parte)           14 (385; 3.6%)   
C340 - Tumor maligno del bronquio principal          21 (144; 14.6%)   
C450 - Mesotelioma de la pleura                       22 (65; 33.8%)   
I051 - Insuficiencia mitral reumática                  7 (21; 33.3%)   

                                                                2017  \
Diagnóstico                                                            
C341 - Tumor maligno del lóbulo superior, bronq...  180 (372; 48.4%)   
Z450 - Asistencia y ajuste de marcapaso cardíaco      71 (718; 9.9%)   
J841 - Otras enfermedades pulmonares interstici...  108 (1666; 6.5%)   
C343 - Tumor maligno del lóbulo inferior, bronq...  118 (258; 45.7%)   
D381 - Tumor de comportamiento incierto o desco...    36 (553; 6.5%)   
J679 - Neumonitis debida a hipersensibilidad a ...   77 (106; 72.6%)   
C780 - Tumor maligno secundario del pulmón           57 (373; 15.3%)   
J848 - Otras enfermedades pulmonares interstici...   25 (113; 22.1%)   
I351 - Insuficiencia (de la válvula) aórtica         37 (318; 11.6%)   
J955 - Estenosis subglótica consecutiva a proce...   53 (167; 31.7%)   
I080 - Trastornos de las válvulas mitral y aórtica    38 (96; 39.6%)   
I081 - Trastornos de las válvulas mitral y tric...    29 (51; 56.9%)   
J47X - Bronquiectasia                                53 (377; 14.1%)   
E848 - Fibrosis quística con otras manifestaciones    26 (58; 44.8%)   
C782 - Tumor maligno secundario de la pleura         29 (147; 19.7%)   
I340 - Insuficiencia (de la vál

In [44]:
resumen_egresos_y_posiciones_relevantes = resumen_egresos_y_posiciones_relevantes.reset_index()

In [45]:
visualize.add_dataframes_to_powerpoint(
    [resumen_egresos_y_posiciones_relevantes],
    "../reports/presentacion.pptx",
    font_size=8,
    cell_width=33,
    cell_height=0.5,
    font_family="Open Sans",
    max_cell_characters=100,
)

Para hacer un contorl de lo que se calculo, se observará una muestra de 10 hospitales - diagnósticos
distintos


In [46]:
display(rankings_nacionales.sample(10))

shape: (10, 27)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ ANO_EGRES ┆ ESTABLECI ┆ GLOSA_EST ┆ Capítulo  ┆ … ┆ %_grd_n_e ┆ ranking_i ┆ total_int ┆ %_intern │
│ O         ┆ MIENTO_SA ┆ ABLECIMIE ┆ ---       ┆   ┆ gresos    ┆ nterno_n_ ┆ erno_n_eg ┆ o_n_egre │
│ ---       ┆ LUD       ┆ NTO_SALUD ┆ str       ┆   ┆ ---       ┆ egresos   ┆ resos     ┆ sos      │
│ i64       ┆ ---       ┆ ---       ┆           ┆   ┆ f64       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ i64       ┆ str       ┆           ┆   ┆           ┆ u32       ┆ u32       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ 2013      ┆ 104104    ┆ Hospital  ┆ Cap.15    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Dr.       ┆ EMBARAZO, ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Manuel    ┆ PARTO Y   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Magalhaes ┆ PUERPE…   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Me…       ┆           ┆   ┆           ┆           ┆           ┆          │
│ 2014      ┆ 119207    ┆ Clínica   ┆ Cap.19    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Bio Bio   ┆ LESIONES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ TRAUMÁTIC ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ AS, EN…   ┆   ┆           ┆           ┆           ┆          │
│ 2002      ┆ 113100    ┆ Hospital  ┆ Cap.09    ┆ … ┆ 0.04918   ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Barros    ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Luco      ┆ DES DEL   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Trudeau   ┆ APARATO…  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ (Sa…      ┆           ┆   ┆           ┆           ┆           ┆          │
│ 2004      ┆ 121202    ┆ Clínica   ┆ Cap.19    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Alemana   ┆ LESIONES  ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ de Temuco ┆ TRAUMÁTIC ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ AS, EN…   ┆   ┆           ┆           ┆           ┆          │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ 2007      ┆ 125100    ┆ Hospital  ┆ Cap.01    ┆ … ┆ 0.002331  ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Regional  ┆ CIERTAS   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ (Coihaiqu ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ e)        ┆ DES INF…  ┆   ┆           ┆           ┆           ┆          │
│ 2017      ┆ 111200    ┆ Hospital  ┆ Cap.01    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ Clínico   ┆ CIERTAS   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ Universid ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆ ad Cat…   ┆ DES INF…  ┆   ┆           ┆           ┆           ┆          │
│ 2015      ┆ 122101    ┆ Hospital  ┆ Cap.09    ┆ … ┆ null      ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de Corral ┆ ENFERMEDA ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ DES DEL   ┆   ┆           ┆           ┆           ┆          │
│           ┆           ┆           ┆ APARATO…  ┆   ┆           ┆           ┆           ┆          │
│ 2015      ┆ 111195    ┆ Hospital  ┆ Cap.10    ┆ … ┆ 0.00463   ┆ null      ┆ null      ┆ null     │
│           ┆           ┆ de        ┆ ENFERME

En la muestra anterior se puede observar las métricas previamente calculada, y el ranking
respectivo que tiene cada hospital para diagnósticos en cada año específico.


In [47]:
rankings_nacionales.to_pandas().to_csv(
    "../data/interim/ranking_nacional_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

rankings_hospital_analizado.to_pandas().to_csv(
    "../data/interim/ranking_torax_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

## Metricas para personas sobre 14 años

En este análisis se quiere ver el ranking de egresos pero solamente tomando en cuenta a personas
mayores a 14 años. Esto, para dilucidar si el hospital del tórax es altamente relevante para tratar
patologías congénitas de adultos.


In [48]:
df_nacional_mayores_a_14 = df_nacional.filter(pl.col("EDAD_A_OS") > 14)

In [49]:
metricas_sobre_14 = build_features.obtener_metricas_egresos(
    df_nacional_mayores_a_14, AGRUPACION
).collect()

In [50]:
variables_a_rankear = ["n_egresos"]
subgrupo_del_ranking = ["ANO_EGRESO", "DIAG1"]

rankings_nacionales_sobre_14 = build_features.agregar_ranking_estratos(
    metricas_sobre_14, dict_estratos, variables_a_rankear, subgrupo_del_ranking
)

diags_congenitos_relevantes = rankings_nacionales_sobre_14.filter(
    (pl.col("DIAG1").is_in(["Q230", "Q211", "Q231"])) & (pl.col("ANO_EGRESO") == 2019)
)

In [51]:
diags_congenitos_relevantes.to_pandas().to_csv(
    "../data/interim/diags_congenitas_sobre_14_egresos.csv",
    sep=";",
    decimal=".",
    encoding="latin-1",
    index=False,
)

## 3. Análisis de procedimientos DEIS

En este análisis se quieren responder las siguientes preguntas:

1. ¿Cuál es el procedimiento más frecuente dentro de los pacientes que se les realizó un procedimiento
   quirúrgico?


In [52]:
proced = (
    df_nacional.filter(pl.col("INTERV_Q") == 1)
    .groupby(["ANO_EGRESO", "ESTABLECIMIENTO_SALUD", "GLOSA_PROCED_PPAL"])
    .agg(pl.count())
).collect()

C:\Users\ppizarro\AppData\Local\Temp\ipykernel_20320\3397463805.py:3: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.
  .groupby(["ANO_EGRESO", "ESTABLECIMIENTO_SALUD", "GLOSA_PROCED_PPAL"])


In [53]:
display(proced)

shape: (10_757, 4)
┌────────────┬───────────────────────┬───────────────────────────────────┬───────┐
│ ANO_EGRESO ┆ ESTABLECIMIENTO_SALUD ┆ GLOSA_PROCED_PPAL                 ┆ count │
│ ---        ┆ ---                   ┆ ---                               ┆ ---   │
│ i64        ┆ i64                   ┆ str                               ┆ u32   │
╞════════════╪═══════════════════════╪═══════════════════════════════════╪═══════╡
│ 2001       ┆ 112100                ┆ null                              ┆ 13867 │
│ 2001       ┆ 101100                ┆ null                              ┆ 4816  │
│ 2001       ┆ 112103                ┆ null                              ┆ 1143  │
│ 2001       ┆ 109101                ┆ null                              ┆ 4618  │
│ …          ┆ …                     ┆ …                                 ┆ …     │
│ 2020       ┆ 200486                ┆ Resonancia Magnética  de columna… ┆ 1     │
│ 2020       ┆ 112249                ┆ Doppler simple de vasos periferi… ┆ 2     │
│ 2020       ┆ 117225                ┆ E.C.G. continuo (test Holter o s… ┆ 1     │
│ 2020       ┆ 107206                ┆ Luxaciones de articulaciones  ma… ┆ 1     │
└────────────┴───────────────────────┴───────────────────────────────────┴───────┘

Los resultados muestran el desglose por año y recinto hospitalario junto a la cantidad de
procedimientos realizados en cada intervención quirúrgica. Sin embargo, se puede observar que
existe una cantidad importante de casos donde la glosa del procedimiento fue nula.

Ahora, se quiere calcular cuántas intervenciones quirúrgicas carecen de una glosa de procedimiento.


In [54]:
display(proced.groupby("GLOSA_PROCED_PPAL").agg(pl.col("count").sum()).sort("count"))

C:\Users\ppizarro\AppData\Local\Temp\ipykernel_20320\989951252.py:1: DeprecationWarning: `groupby` is deprecated. It has been renamed to `group_by`.
  display(proced.groupby("GLOSA_PROCED_PPAL").agg(pl.col("count").sum()).sort("count"))


shape: (422, 2)
┌───────────────────────────────────┬──────────┐
│ GLOSA_PROCED_PPAL                 ┆ count    │
│ ---                               ┆ ---      │
│ str                               ┆ u32      │
╞═══════════════════════════════════╪══════════╡
│ Resonancia Magnética de Antebraz… ┆ 1        │
│ Spect - Tomografía por emisión f… ┆ 1        │
│ Respuesta respiratoria al co2     ┆ 1        │
│ Radiografía retroalveolar y Bite… ┆ 1        │
│ …                                 ┆ …        │
│ Instilacion vesical (incluye col… ┆ 7685     │
│ Código no existe actualmente      ┆ 11155    │
│ Apoyo fluoroscópico a procedimie… ┆ 15775    │
│ null                              ┆ 12382642 │
└───────────────────────────────────┴──────────┘

Por lo tanto, 12382642 de todas las intervenciones quirúrgicas carecen de una glosa de procedimiento.


In [55]:
proced.write_excel("../data/interim/conteo_procedimientos.xlsx")